src/compiler/types.ts

```diff
@@ -5119,7 +5121,7 @@ export interface TypeChecker {
     /** @internal */ getPromiseLikeType(): Type;
     /** @internal */ getAsyncIterableType(): Type | undefined;
 
-    /** @internal */ isTypeAssignableTo(source: Type, target: Type): boolean;
+    isTypeAssignableTo(source: Type, target: Type): boolean;

```

Comment:
I think if we leave this one out and discuss #9879 separately, the rest should be pretty uncontroversial.